pkg/kstatus/polling/statusreaders/common.go

```diff
@@ -209,3 +209,10 @@ func errIdentifierToResourceStatus(err error, identifier object.ObjMetadata) (*e
 		Error:      err,
 	}, nil
 }
+
+func errWouldExceedContextDeadline(err error) bool {
+	if e := errors.Unwrap(err); e != nil {
+		return e.Error() == "rate: Wait(n=1) would exceed context deadline"
```

Comment:
This is very fragile. There should be at least a test for this. Ideally the wait package should have a type for this.

Also, this doesn't feel like the right spot for this logic.

Reply:
Agree, ideally `rate` package would return a typed error here

Reply:
Added a test.

> Also, this doesn't feel like the right spot for this logic.

Do you have a suggestion what would be a better place?

Reply:
Maybe somewhere up the call stack. I'm not a maintainer of this project, just a user, so this may be fine with maintainers.